net/dnscache/dnscache.go

```diff
@@ -457,9 +457,7 @@ func TLSDialer(fwd DialContextFunc, dnsCache *Resolver, tlsConfigBase *tls.Confi
 			}
 		}()
 		go func() {
-			err := tlsConn.Handshake()
-			handshakeTimeoutCancel()
```

Comment:
ah! I see. it made the L452 goroutine readable and send to errc first sometimes.

Reply:
Yup, exactly.